### ISSUE TITLE
Runtime failure with quadmath

### ISSUE THREAD

**Issue by Tux** [DBD-SQLite/DBD-SQLite#65]
```
This is perl 5, version 32, subversion 1 (v5.32.1) built for x86_64-linux-thread-multi-quadmath
```
Before:
```
t/58_see_if_its_a_number_and_explicit_binding.t ....... 1/? panic: quadmath invalid format "%.2f" at /data/home/merijn/.cpan/build/DBD-SQLite-1.66-0/blib/lib/DBD/SQLite.pm line 233.
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 1.
t/58_see_if_its_a_number_and_explicit_binding.t ....... Dubious, test returned 255 (wstat 65280, 0xff00)
All 1 subtests passed
```
Patch:
```
--- dbdimp.c.org        2021-01-25 17:46:32.061708809 +0100
+++ dbdimp.c    2021-01-25 17:42:53.579725176 +0100
@@ -404,7 +404,11 @@ sqlite_is_number(pTHX_ const char *v, in
         if (!_sqlite_atoi64(v, &iv)) return 1;
     }
     if (sql_type != SQLITE_INTEGER) {
-        sprintf(format, (has_plus ? "+%%.%df" : "%%.%df"), precision);
+#ifdef USE_QUADMATH
+        sprintf(format, (has_plus ? "+%%.%dQf" : "%%.%dQf"), precision);
+#else
+        sprintf(format, (has_plus ? "+%%.%df"  : "%%.%df" ), precision);
+#endif
         if (strEQ(form(format, atof(v)), v)) return 2;
     }
     return 0;
```
After
```
All tests successful.

Test Summary Report
-------------------
t/58_see_if_its_a_number_and_explicit_binding.t     (Wstat: 0 Tests: 8 Failed: 0)
  TODO passed:   2
Files=113, Tests=3834, 21 wallclock secs ( 0.50 usr  0.11 sys +  8.13 cusr  1.05 csys =  9.79 CPU)
Result: PASS
```

Do you want a PR?

**Comment by Leont** [comment]
Instead of this `#ifdef USE_QUADMATH`, you should be able to use
```
sprintf(format, (has_plus ? "+%%.%d" NVff : "%%.%d" NVff), precision);
```

**Comment by charsbar** [comment]
Thank you both. Applied Leon's patch and released 1.67_03.
